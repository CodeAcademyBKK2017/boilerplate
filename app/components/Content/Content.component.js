// import Icon from 'react-native-vector-icons/Foundation';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Content.style';
import {Text, TextInput, View} from 'react-native';

export default class TextArea extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View style={styles.headContent}>
          <Text style={styles.text}>Please type your note below</Text>
          {/* <Icon name='clipboard-notes' size={20} /> */}
        </View>
        <TextInput style={styles.textArea} 
          multiline={true} 
          underlineColorAndroid='transparent'
          onChangeText={this.props.onContentChange}  
          value={this.props.value}/>
      </View>
    );
  }
}
TextArea.propTypes = {
  onContentChange: PropTypes.func,
  value: PropTypes.string
};
