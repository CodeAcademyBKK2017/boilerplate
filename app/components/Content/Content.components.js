import React, {Component} from 'react';
import {

  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

class Content extends Component {
  render () {
    return (
      <View style={styles.conView}>
        <Text style={styles.welcome}>
         Please type your note below
        </Text>
        <TextInput
          style={styles.textin}
          // onChangeText={this.state.nans}
          multiline = {true}
          underlineColorAndroid = 'transparent'
          numberOfLines = {4}   
          placeholder='-Understand how react-native works.'
        />
      </View>
    );
  }
}
const styles = StyleSheet.create({
  welcome: {
    fontSize: 12,
    // textAlign: 'center',
    margin: 10
   
  },
  conView: {
    flex: 1,
    marginTop: 10
  },
  textin: {
    // width: 300,
    // height: 300,
    borderColor: 'gray',
    borderWidth: 1,
    flex: 1,
    backgroundColor: '#fff'

  }
});
export default Content;