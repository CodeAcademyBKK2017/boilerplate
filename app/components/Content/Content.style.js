import React, {StyleSheet} from 'react-native';
  
export default StyleSheet.create({
  welcome: {
    fontSize: 16,
    textAlign: 'center',
    margin: 10,

    fontStyle: 'italic'
  },
  textArea: {
    width: '100%',
    borderColor: 'gray',
    borderWidth: 1,
    backgroundColor: '#fff',
    padding: 5,
    color: '#ababab',
    borderRadius: 3
  },
  content: {
    flexDirection: 'row',
    margin: 20
  },
  conArea: {
    flex: 1,
    flexDirection: 'row',
    marginLeft: 30,
    marginRight: 30
  },
  title: {
    flex: 1,
    flexDirection: 'row',
    margin: 30
  },
  noteLeft: {
    flex: 1
  },
  noteRight: {
    flex: 0
  },
  note: {
    fontWeight: '600'
  }
});