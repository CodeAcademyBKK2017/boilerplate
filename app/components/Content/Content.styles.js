import React, {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    flex: 1,
    paddingBottom: 10
  },
  title: {
    fontSize: 13,
    color: '#444',
    fontStyle: 'italic'
  },
  textInput: {
    flex: 1,

    marginVertical: 10,
    paddingVertical: 15,
    paddingHorizontal: 12,

    borderWidth: 1,
    borderColor: '#c4c4c5',
    borderRadius: 3,
    
    backgroundColor: '#ffffff',
    textAlignVertical: 'top'
  }
});