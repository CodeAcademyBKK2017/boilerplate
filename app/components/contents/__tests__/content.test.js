import 'react-native';
import Content from '../content.component';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// import {shallow} from 'enzyme';

describe('Content', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Content />
    );
    expect(tree).toBeDefined();
  });
  it('Content: renders correctly', () => {
    const tree = renderer.create(<Content />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
