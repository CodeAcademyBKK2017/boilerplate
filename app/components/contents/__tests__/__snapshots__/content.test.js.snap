// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Content: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 9,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "fontStyle": "italic",
        "marginBottom": 5,
        "textAlign": "left",
      }
    }
  >
    Please type your note below.
  </Text>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    style={
      Object {
        "backgroundColor": "#F5FCFF",
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexGrow": 1,
        "width": 350,
      }
    }
    value=""
  />
</View>
`;
